CONTRIBUTING.md

```diff
@@ -800,45 +744,280 @@ Try restricting ifdefs in the header files, or in the beginning of
 the C files.
 
 
-## Advanced topics
+## Suggested git / GitHub workflow
 
-This section is not for new contributors, but rather contains more specific guides which are useful to link to in some circumstances.
+For very small / simple changes, you can open a PR by clicking the edit
+buttons in the GitHub UI. For most changes, it is preferable to work locally,
+using git. This is a suggested outline for those unfamiliar with working with
+GitHub. It does, however, assume a familiarity with the basics of using git
+so it does not attempt to provide every detail of that.
 
 
-### Git
+### Forking
 
+You will be working with three git repositories: the central one on Github,
+your fork on Github and your local clone.
 
-#### How to rebase a branch
+Within Github, make your fork of the central CFEngine repository
+using the fork icon on the
+[Github CFEngine repository page](https://github.com/cfengine/core).
+Your fork lives at `https://github.com/your-username/core`.
+
+
+### Cloning
+
+Clone your fork, there is a **"Clone or Download"** button, which shows the
+necessary information. For example:
+
+```
+$ git clone --recursive https://github.com/your-username/core
+$ cd core
+```
+
+(If you have set up your GitHub profile with your SSH key, you can use the SSH
+URLs instead, which might be more convenient).
+
+Replace `your-username` with your GitHub username, here and later.
+`--recursive` is needed to clone submodules, without it, you would have to run
+these commands separately:
+
+```
+$ git submodule init
+$ git submodule update
+```
+
+
+### Set up remote references
+
+Now, add a `remote` reference to the `upstream` repository:
+
+```
+$ git remote add upstream https://github.com/cfengine/core
+```
+
+The result should look like this:
+
+```
+$ git remote -v
+origin	https://github.com/olehermanse/core (fetch)
+origin	https://github.com/olehermanse/core (push)
+upstream	https://github.com/cfengine/core (fetch)
+upstream	https://github.com/cfengine/core (push)
+```
+
+`origin` is your fork. This is the default for pushing and pulling.
+`upstream` is the central repository, managed by the CFEngine GitHub
+organization. You (probably) don't have access to push to `upstream`.
+Using these names are just a convention, you could change them, but we don't
+recommend it.
+
+
+### Branching
+
+Never work in the `master` branch; always use a separate branch for your topic.
+Prepare your topic on this topic-focussed branch.
+When it is ready, push it to your Github fork.
+
+```
+$ git checkout master
+$ git checkout -b mytopic
```

Comment:
Perhaps `git checkout -b mytopic upstream/master`

Reply:
I don't think this makes it easier to understand or read, and this isn't really a comprehensive git tutorial, so I'll leave it as is.